### PR TITLE
FEATURE: show assigned user on search results

### PR DESCRIPTION
**Pull request by techAPJ**
This commit allows discourse-assign plugin to show assigned users on search result topic list.

Accompanying PR on meta: https://github.com/discourse/discourse/pull/11518

### CODE REVIEW

File: plugin.rb [hunk 1]
```diff
@@ -182,6 +183,28 @@
     end
   end
 
+  if Search.respond_to?(:on_preload)
+    Search.on_preload do |search, search_query|
```
Comment (CvX) [inline plugin.rb:187]: Mirroring the comment from the other PR, are `search` and `search_query` the best names for those variables?

File: plugin.rb [hunk 2]
```diff
@@ -182,6 +183,28 @@
...
+      if SiteSetting.assign_enabled?
+        can_assign = search_query.guardian && search_query.guardian.can_assign?
+        allowed_access = SiteSetting.assigns_public || can_assign
+
+        if allowed_access && search.posts.length > 0
+          assigned_user_ids = search.posts.map(&:topic).map { |topic| topic.custom_fields[TopicAssigner::ASSIGNED_TO_ID] }.compact.uniq
+          assigned_users = {}
+          User.where(id: assigned_user_ids).each do |user|
+            assigned_users[user.id] = user
+          end
```
Comment (CvX) [inline plugin.rb:197]:
```suggestion
          assigned_users = User.where(id: assigned_user_ids).index_by(&:id)
```

**Review by CvX — CHANGES_REQUESTED** [review]

File: plugin.rb (hunk 1 above)
Comment (techAPJ) [reply plugin.rb]: Makes sense, updated variable names.

File: plugin.rb (hunk 2 above)
Comment (techAPJ) [reply plugin.rb]: Thanks for the tip! 👍

**Review by CvX — APPROVED** [review]

**Pull request merged**